docs/guides/user-containers/readme.md

```diff
@@ -262,16 +262,106 @@ allowed per workflow.
     GFS2 filesystems have special considerations since the mount directory contains directories for
     every compute node. See [GFS2 Index Mounts](#gfs2-index-mounts) for more info.
 
+### Targeting Nodes
+
+For container directives, compute nodes must be assigned to the workflow. The NNF software will
+trace the compute nodes back to their local NNF nodes and the containers will be executed on those
+NNF nodes. The act of assigning compute nodes to your container workflow instructs the NNF software
+to select the NNF nodes that run the containers.
+
+For the `jobdw` directive that is included above, the servers (i.e. NNF nodes) must also be assigned
+along with the computes.
+
 ## Running a Container Workflow
 
-TODO: Once the container exit strategy is complete, come back to this section to detail:
+Once the workflow is created, the WLM progresses it through the following states. This is a quick
+overview of the container-related behavior that occurs:
+
+- Proposal: Verify storages are provided according to the container profile.
+- Setup: If applicable, request ports from NnfPortManager.
+- DataIn: No container related activity.
+- PreRun: Appropriate `MPIJob` or `Job(s)` are created for the workflow. In turn, user containers
+are created and launched by Kubernetes. Containers are expected to start in this state.
+- PostRun: Once in PostRun, user containers are expected to complete (non-zero exit)
+successfully.
+- DataOut: No container related activity.
+- Teardown: Ports are released; `MPIJob` or `Job(s)` are deleted, which in turn deletes the user
+containers.
+
+The two main states of a container workflow (i.e. PreRun, PostRun) are discussed further in the
+following sections.
+
+### PreRun
+
+In PreRun, the containers are created and expected to start. Once the containers reach a
+non-initialization state (i.e. Running), the containers are considered to be started and the
+workflow can advance.
+
+By default, containers are expected to start within 60 seconds. If not, the workflow reports an
+Error that the containers cannot be started. This value is configurable via the
+`preRunTimeoutSeconds` field in the container profile.
+
+To summarize the PreRun behavior:
+
+- If the container starts successfully (running), transition to Ready:true.
+- If the container fails to start, transition to the Error State.
+- If the container is initializing and has not started after `preRunTimeoutSeconds` seconds,
+terminate the container and transition to the Error State.
+
+#### Init Containers
 
-- PreRun Behavior
-- PostRun Behavior
-- Teardown
+Containers have some additional initialization that occurs using [Init
+Containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/).  These containers
+must run to completion before the main container can start. The NNF Software injects Init
```

Comment:
This:
"The NNF Software injects Init Containers into the container specification"
is the key statement in this paragraph, and sub-section.  It should be the first sentence.

Reply:
I'll reword this